cirq/neutral_atoms/neutral_atom_devices.py

```diff
@@ -0,0 +1,426 @@
+        # Also verify that gates of the same type are equal
+        for op in operations:
+            if isinstance(op, (ops.GateOperation,
+                               ops.ParallelGateOperation)):
+                if isinstance(op.gate, ops.ZPowGate):
+                    if num_parallel_z == 0:
+                        refZgate = op.gate
+                    else:
+                        if refZgate != op.gate:
+                            raise ValueError("Non-identical Parallel Z gates")
+                    num_parallel_z += len(op.qubits)
+
+                if isinstance(op.gate, (ops.XPowGate,
+                                        ops.YPowGate,
+                                        ops.PhasedXPowGate)):
+                    if num_parallel_xy == 0:
+                        refXYgate = op.gate
+                    else:
+                        if refXYgate != op.gate:
+                            raise ValueError("Non-identical Parallel XY gates")
+                    num_parallel_xy += len(op.qubits)
+
+                if isinstance(op.gate, (ops.CNotPowGate,
+                                        ops.CZPowGate,
+                                        ops.CCXPowGate,
+                                        ops.CCZPowGate)):
+                    if len(controlled_qubits_lists) == 0:
+                        refCgate = op.gate
+                    else:
+                        if refCgate != op.gate:
+                            raise ValueError("Non-identical Parallel Controlled"
+                                             " Gates")
+                    controlled_qubits_lists.append(op.qubits)
+                if isinstance(op.gate, ops.MeasurementGate):
+                    has_measurement = True
+
+        if len(controlled_qubits_lists) > 0:
+            if (sum([len(l) for l in controlled_qubits_lists]) >
+                    self._max_parallel_c):
+                raise ValueError("Too many qubits acted on by controlled gates")
+            if num_parallel_xy > 0 or num_parallel_z > 0:
+                raise ValueError("Can't perform non-controlled operations at"
+                                 " same time as controlled operations")
+            if self.are_qubit_lists_too_close(*controlled_qubits_lists):
+                raise ValueError("Interacting controlled operations")
+
+        if num_parallel_z > self._max_parallel_z:
+            raise ValueError("Too many simultaneous Z gates")
+
+        if (num_parallel_xy > self._max_parallel_xy and
+                num_parallel_xy != len(self.qubits)):
+            raise ValueError("Bad number of simultaneous XY gates")
+
+        if has_measurement:
+            if (len(controlled_qubits_lists) > 0 or num_parallel_z > 0 or
+                    num_parallel_xy > 0):
+                raise ValueError("Measurements can't be simultaneous with other"
+                                 " operations")
+
+    def are_qubit_lists_too_close(self,
```

Comment:
Add a docstring, or make this method private (by prefixing with an underscore).

Reply:
Since this method is analagous to the private xmon method that checks for interacting gates I'll make this private.